packages/smart-wallet/src/proposals/upgrade-walletFactory-proposal.js

```diff
@@ -0,0 +1,128 @@
+// @ts-check
+import { E } from '@endo/far';
+import { makeMarshal } from '@endo/marshal';
+import { allValues } from '@agoric/internal';
+
+console.warn('upgrade-walletFactory-proposal.js module evaluating');
+
+const { Fail } = assert;
+
+// vstorage paths under published.*
+const WALLET_STORAGE_PATH_SEGMENT = 'wallet';
+const BOARD_AUX = 'boardAux';
+
+const marshalData = makeMarshal(_val => Fail`data only`);
+
+/**
+ * @param { BootstrapPowers } powers
+ *
+ * @param {object} config
+ * @param {{ walletFactoryRef: VatSourceRef & { bundleID: string } }} config.options
+ */
+export const upgradeWalletFactory = async (
+  {
+    consume: {
+      contractKits,
+      governedContractKits,
+      chainStorage,
+      walletBridgeManager: walletBridgeManagerP,
+    },
+    instance: {
+      consume: { walletFactory: wfInstanceP, provisionPool: ppInstanceP },
+    },
+  },
+  config,
+) => {
+  console.log('upgradeWalletFactory: config', config);
+  const { walletFactoryRef } = config.options;
+
+  // console.log('upgradeWalletFactory: awaiting instances etc.');
+  const { wfInstance, ppInstance, walletBridgeManager, storageNode } =
+    await allValues({
+      wfInstance: wfInstanceP,
+      ppInstance: ppInstanceP,
+      walletBridgeManager: walletBridgeManagerP,
+      // @ts-expect-error chainStorage is only falsy in testing
+      storageNode: E(chainStorage).makeChildNode(WALLET_STORAGE_PATH_SEGMENT),
+    });
+  // console.log('upgradeWalletFactory: awaiting contract kits');
+  const { wfKit, ppKit } = await allValues({
+    wfKit: E(contractKits).get(wfInstance),
+    ppKit: E(governedContractKits).get(ppInstance),
+  });
+  // console.log('upgradeWalletFactory: awaiting walletReviver');
+  const walletReviver = await E(ppKit.creatorFacet).getWalletReviver();
+  const newPrivateArgs = harden({
```

Comment:
shouldn't these be the same privateArgs as the running instance?

you can get those from `instancePrivateArgs` in bootstrap promise space.

Reply:
seems like I tried to do that... trying to remember why I went another way...

Reply:
I think it was about `instancePrivateArgs` being only for diagnostic / testing purposes.

But maybe using it in production for upgrade purposes is appropriate?

I wonder about divergence from master, but I think I'll give it a try...

Reply:
hm...

```
2023-09-08T01:05:40.125Z SwingSet: vat: v1: upgradeWalletFactory: upgrading with privateArgs { storageNode: Object [Alleged: ChainStorageNode] {}, walletBridgeManager: Object [Alleged: BridgeScopedManager] {}, walletReviver: Promise [Promise] {} }
2023-09-08T01:05:40.331Z SwingSet: ls: v5: Logging sent error stack (TypeError#1)
2023-09-08T01:05:40.332Z SwingSet: ls: v5: TypeError#1: target has no method getDisplayInfo , has [ 'isMyClock', 'isMyTimerService' ]
```
https://github.com/Agoric/agoric-sdk/actions/runs/6116183603/job/16601003319?pr=8311#step:6:4288